src/components/engagement/handlers/apply-rejected-changeset-to-engagement.handler.ts

```diff
@@ -0,0 +1,137 @@
+import { node, relation } from 'cypher-query-builder';
+import { DateTime } from 'luxon';
+import { ID, ServerException } from '../../../common';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { ACTIVE } from '../../../core/database/query';
+import { rejectChangesetProps } from '../../changeset/reject-changeset-props.query';
+import { ProjectChangeRequestRejectedEvent } from '../../project-change-request/events';
+
+type SubscribedEvent = ProjectChangeRequestRejectedEvent;
+
+@EventsHandler(ProjectChangeRequestRejectedEvent)
+export class ApplyRejectedChangesetToEngagement
+  implements IEventHandler<SubscribedEvent>
+{
+  constructor(
+    private readonly db: DatabaseService,
+    @Logger('engagement:change-request:rejected')
+    private readonly logger: ILogger
+  ) {}
+
+  async handle(event: SubscribedEvent) {
+    this.logger.debug('Rejecting changeset props');
+
+    const changeset = event.changeRequest.id;
+
+    try {
+      // Update project engagement pending changes
+      const engagements = await this.db
+        .query()
+        .match([
+          node('project', 'Project'),
+          relation('out', '', 'changeset', ACTIVE),
+          node('changeset', 'Changeset', { id: changeset }),
+        ])
+        .subQuery((sub) =>
+          sub
+            .with('project')
+            .match([
+              node('project'),
+              relation('out', 'engagementRel', 'engagement', {
+                active: true,
+              }),
+              node('node', 'Engagement'),
+            ])
+            .return('node')
+            .union()
+            .with('project, changeset')
+            .match([
+              node('project'),
+              relation('out', 'engagementRel', 'engagement', {
+                active: false,
+              }),
+              node('node', 'Engagement'),
+              relation('in', 'changesetRel', 'changeset', ACTIVE),
+              node('changeset'),
+            ])
+            .delete('changesetRel')
+            .create([
+              node('node'),
+              relation('in', '', 'previous', {
+                active: true,
+                createdAt: DateTime.local(),
+              }),
+              node('changeset'),
+            ])
```

Comment:
I'm don't think this is right. The previous relation should be applied to properties, not a changeset. This change isn't being removed from the changeset.